### DISCUSSION TITLE
Controls for smart grid charging of the house battery

### DISCUSSION THREAD

**Discussion by maxgrosse1** [marq24/ha-evcc#31] [Ideas]
Would it be possible to add controls for smart grid charging of the house battery, e.g. enable/disable, set energy cost limits, etc.
Best, Max

**Comment by marq24** [comment]
when you would let know me, which evcc api object contain this data (I don't have a appropriate battery here @ home)

**Reply by maxgrosse1** [reply]
Thanks a lot, Matthias. I have found the following [Link](https://github.com/evcc-io/evcc/discussions/16090)

**Reply by UDicke** [reply]
you may also look here https://github.com/marq24/ha-evcc/discussions/24

**Reply by marq24** [reply]
@maxgrosse1 - the Link you mentioned is referring to 'site' property 'batteryGridChargeLimit' - this sensor/number entity is already implemented...

But my understanding is, that you want to control something different?

**Reply by marq24** [reply]
... just again...

- To turn on/off the house_battery charging there is the switch: battery_discharge_control
- To set the energy cost limit there is: battery_grid_charge_limit

so what is then 'etc...' ?! [missing from your point of view?]

**Comment by maxgrosse1** [comment]
I was looking for enabling/disabling grid charging the house battery without setting a specific grid_charge_limit. I am not able to find the mentioned battery_discharge_control, just (in German) "Hausbatterie_Entladesperre" which I understood is just locking the batterie to discharge during fast loading the car battery. Did I misunderstand something?

**Reply by marq24** [reply]
Also erstmal "ja" `Hausbatterie: Entladesperre` ist der Schalter `battery_discharge_control` - was dieser Schalter allerdings in evcc macht und wo Du den innerhalb des evcc Frontend wiederfinden kannst -> keine Ahnung [die evcc-API kennt diesen Key - und man kann ihn lesen & schreiben - also mach' ich das einfach in meiner Integration] - Ich interpretiere diesen Schalter aber ebenso wie Du - die Hausbatterie soll beim laden des Fahrzeuges "auf keinen Fall" genutzt werden... [ggf versucht evcc auch direkt bei der Hausbatterie ein entsprechenden LOCK zu setzen, wenn das die API des Batterieherstellers unterstützt... meiner (Senec) unterstützt es auf jeden Fall nicht]

**Reply by UDicke** [reply]
Evtl. kann ich helfen. Ich vermute es handelt sich bei diesem Sensor nur um den Konfigurationsschalter.
![image](https://github.com/user-attachments/assets/cb0aeb68-770f-4bb3-9cbd-4db5af822f1d)
Die eigentliche Entladesperre wird dann gesetzt wenn Laden aus dem Netz aktiviert wird. Bei meiner Batterie gibt's dafür auch keinen Schalter. Ich habe bei mir eine Entladesperre realisiert indem ich die Notstromkapazität hochsetze, z.B. auf 100%. So macht evcc das auch wenn es keinen Schalter gibt. Bei "SAJ-H2" weiß ich genau das es so funktioniert.Ob der Sensor der Konfigschalter ist, kann ich einfacher prüfen wenn ich zuhause bin.
Habe jetzt doch mit dem Handy geprüft.
Bei evcc_battery_discharge_control handelt es sich um den Schalter zur Konfiguration. Die eigentliche Entladesperre wird nur aktiv wenn ein Auto nach Plan oder im Schnellademodus geladen wird.

**Reply by Detlef21** [reply]
Hi zusammen, gibt es eine Lösung um die Hausbatterie zu bestimmten Zeiten (günstige dyn. Preise) zu sperren um bei hohen Preisen noch etwas von der Batterie zu haben? Der battery_discharge_control Befehl ist es nicht..

**Reply by UDicke** [reply]
nicht über evcc, ich mache das über Home Assistant. Zum Thema
> Der battery_discharge_control Befehl ist es nicht..

hatte ich oben ja schon geschrieben.

Damit die Entladesperre wie oben beschrieben funktioniert, muss es eine Batterie sein die diesen Modus unterstützt.

**Reply by Detlef21** [reply]
> nicht über evcc, ich mache das über Home Assistant.

UDicke, wie genau im Home Assitant? Steuerst du aber über die EVC-Intergration, oder? Bei mir ist es eine Luna 2000 S1 Batterie.. Danke!

**Comment by UDicke** [comment]
Diese Batterie finde ich in der evcc Doku leider nicht.
Ich steuer meine SAJ-H2 bzw Ampere.storage.pro direkt über Modbus und über die HA evcc Integration, z.B. Preislimit für Laden aus dem Netz setzen. Die Entladesperre steuer ich über das Setzen der Notstromkapazität.

**Comment by btota** [comment]
Hi, thx for great evcc integration! I would like to control the battery load power, because my experiences with evcc are that it starts charging with max load power (sungrow 12.8kW/8.76kWh) and it triggers depending on how long the battery was loading by max power the  "Zwangsentladung" balancing mode which means discharging the battery completely to NULL. So for now I use an own automation for smart charging and using the register for "forced charge discharge power in W" to set/update the value for max load power to e.c. 4.5kWh. 
I tell u this because I didn´t find an rest sensor or api in the docs in evcc which gives control on this - or do you have a clue?
